WikipediaUnitTests/Code/ArticleCacheReadingTests.swift

```diff
@@ -0,0 +1,284 @@
+            case "app://upload.wikimedia.org/wikipedia/en/thumb/a/a4/Flag_of_the_United_States.svg/960px-Flag_of_the_United_States.svg.png":
+                imageExpectation.fulfill()
+            default:
+                break
+            }
+        }
+        
+        UIApplication.shared.keyWindow?.rootViewController = basicVC
+        let _ = basicVC.view
+        
+        wait(for: [htmlExpectation], timeout: 3)
+        wait(for: [imageExpectation], timeout: 3)
+        wait(for: [cssExpectation], timeout: 3)
+        
+        LSNocilla.sharedInstance().stop()
+    }
+    
+    func testBasicNetworkNotModifiedWithCachedArticle() {
+        
+        LSNocilla.sharedInstance().start()
+        stub304Responses()
+        ArticleTestHelpers.writeCachedPiecesToCachingSystem()
+        
+        let basicVC = BasicCachingWebViewController()
+        
+        let htmlExpectation = expectation(description: "Waiting for html content to return")
+        let imageExpectation = expectation(description: "Waiting for image load to return")
+        let cssExpectation = expectation(description: "Waiting for css load to return")
+        
+        basicVC.didReceiveDataCallback = { urlSchemeTask, data in
+            
+            guard let urlString = urlSchemeTask.request.url?.absoluteString else {
+                XCTFail("Unable to determine urlString from scheme task")
+                return
+            }
+            
+            switch urlString {
+            case "app://en.wikipedia.org/api/rest_v1/page/mobile-html/United_States":
+                
+                htmlExpectation.fulfill()
+                
+                let htmlString = String(decoding: data, as: UTF8.self)
+                let trimmedHTML = String(htmlString.filter { !"\n\t\r".contains($0) })
+                XCTAssertEqual(trimmedHTML, "<!DOCTYPE html><html><head><link rel=\"stylesheet\" href=\"//en.wikipedia.org/api/rest_v1/data/css/mobile/site\"></head><body><p>Testing Cached</p><img src=\"//upload.wikimedia.org/wikipedia/en/thumb/a/a4/Flag_of_the_United_States.svg/960px-Flag_of_the_United_States.svg.png\"></body></html>")
+                
+            case "app://en.wikipedia.org/api/rest_v1/data/css/mobile/site":
+                
+                cssExpectation.fulfill()
+                
+                let cssString = String(decoding: data, as: UTF8.self)
+                let trimmedCSS = String(cssString.filter { !"\n\t\r".contains($0) })
+                XCTAssertEqual(trimmedCSS, "body {background-color: red;}", "Unexpected basic HTML content")
+            case "app://upload.wikimedia.org/wikipedia/en/thumb/a/a4/Flag_of_the_United_States.svg/960px-Flag_of_the_United_States.svg.png":
+                imageExpectation.fulfill()
+            default:
+                break
+            }
+        }
+        
+        UIApplication.shared.keyWindow?.rootViewController = basicVC
+        let _ = basicVC.view
```

Comment:
I thought this was needed to properly trigger `viewDidLoad()` from unit tests but it could be my information is woefully out of date. I removed this in a commit and unit tests seem to run fine.